smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/HttpUriConflictValidator.java

```diff
@@ -37,45 +48,145 @@
                 .filter(shape -> shape != operation)
                 .flatMap(shape -> Trait.flatMapStream(shape, HttpTrait.class))
                 .filter(other -> other.getRight().getMethod().equals(method))
                 .filter(other -> other.getRight().getUri().conflictsWith(pattern))
-                .map(other -> String.format("`%s` (%s)", other.getLeft().getId(), other.getRight().getUri()))
-                .sorted()
-                .collect(Collectors.joining(", "));
+                .forEach(other -> {
+                    // Now that we know we have a conflict, determine whether it is allowable or not.
+                    if (isAllowableConflict(model, operation, other.getLeft())) {
+                        allowableConflicts.add(Pair.of(other.getLeft().getId(), other.getRight().getUri()));
+                    } else {
+                        conflicts.add(Pair.of(other.getLeft().getId(), other.getRight().getUri()));
+                    }
+                });
+
+        // Non-allowable conflicts get turned into ERRORs, they must be resolved to pass validation.
+        List<ValidationEvent> events = new ArrayList<>();
+        if (!conflicts.isEmpty()) {
+            events.add(error(operation, formatConflicts(pattern, conflicts)));
+        }
+
+        // Allowable conflicts get turned into DANGERs, which must at least be acknowledged with a suppression.
+        if (!allowableConflicts.isEmpty()) {
+            events.add(danger(operation, formatConflicts(pattern, allowableConflicts)));
+        }
+
+        return events;
+    }
+
+    private boolean isAllowableConflict(Model model, OperationShape operation, OperationShape otherOperation) {
+        UriPattern uriPattern = operation.getTrait(HttpTrait.class).get().getUri();
+        UriPattern otherUriPattern = otherOperation.getTrait(HttpTrait.class).get().getUri();
+
+        List<Pair<Segment, Segment>> conflictingSegments = uriPattern.getConflictingSegments(otherUriPattern);
+
+        // If there aren't any conflicting segments that means the uris are identical, which is not allowable.
+        if (conflictingSegments.isEmpty()) {
+            return false;
+        }
+
+        Map<String, Pattern> labelPatterns = getLabelPatterns(model, operation);
+        Map<String, Pattern> otherLabelPatterns = getLabelPatterns(model, otherOperation);
 
-        if (conflicts.isEmpty()) {
-            return Optional.empty();
+        return conflictingSegments.stream()
+                // Only allow conflicts in cases where one of the segments is static and the other is a label.
+                .filter(conflict -> conflict.getLeft().isLabel() != conflict.getRight().isLabel())
+                // Only allow the uris to conflict if every conflicting segment is allowable.
+                .allMatch(conflict -> {
+                    Pattern pattern;
+                    String staticSegment;
+
+                    if (conflict.getLeft().isLabel()) {
```

Comment:
Seems like this could be simplified if you first resolve which of the two are the label, place that result in a variable, and then extract the pattern and static segment from that.

Reply:
I don't think that simplifies anything. I need to know which pattern map to search.